src/App.tsx

```diff
@@ -1,24 +1,253 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { Notification } from './components/Notification';
+
+import { Todo } from './types/Todo';
+import { FilterBy } from './types/FilterBy';
+import { Errors } from './types/Errors';
 
-const USER_ID = 0;
+import { client } from './utils/fetchClient';
+import * as utils from './utils/utils';
+
+const USER_ID = 11246;
+const BASE_ADD_URL = '/todos';
+const URL_GET = `?userId=${USER_ID}`;
+const DELAY_ERROR = 3000;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [hasTodoLoaderIds, setHasTodoLoaderIds] = useState<number[]>([]);
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.all);
+  const [error, setError] = useState<Errors | null>(null);
+
+  const addLoader = (id: number) => {
+    setHasTodoLoaderIds(currentIds => [...currentIds, id]);
+  };
+
+  const deleteLoader = (id: number) => {
+    setHasTodoLoaderIds(currentIds => [...currentIds].filter(
+      hasLoaderId => hasLoaderId !== id,
+    ));
+  };
+
+  const changeAndToggleTodo = (
+    id: number,
+    field: keyof Todo,
+    value: Todo[typeof field],
+  ) => {
+    const indexOfMutableTodo = todos.findIndex(todo => todo.id === id);
+
+    switch (field) {
+      case 'completed':
+        setTodos(currentTodos => {
+          const copyTodos = [...currentTodos];
+
+          copyTodos[indexOfMutableTodo].completed
+            = !value;
+
+          return copyTodos;
+        });
+        break;
+      case 'title':
+        setTodos(currentTodos => {
+          const copyTodos = [...currentTodos];
+
+          copyTodos[indexOfMutableTodo].title
+            = `${value}`;
+
+          return copyTodos;
+        });
+        break;
+      default:
+        break;
+    }
+  };
+
+  const applyFilter = (filter: FilterBy) => {
+    setFilterBy(filter);
+  };
+
+  const addTodo = async (title: string) => {
+    if (title) {
+      const newTodo = {
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      setHasTodoLoaderIds([0]);
+      setTempTodo({
+        ...newTodo,
+        id: 0,
+      });
+
+      try {
+        const addedTodo = await client.post<Todo>(BASE_ADD_URL, newTodo);
+
+        setTodos(currentTodos => [
+          ...currentTodos,
+          addedTodo,
+        ]);
+      } catch (caughtError) {
```

Comment:
```suggestion
      } catch {
```
you may not use it at all